src/cdk-experimental/drag-drop/drag.ts

```diff
@@ -0,0 +1,493 @@
+  ViewContainerRef,
+  EmbeddedViewRef,
+} from '@angular/core';
+import {CdkDragHandle} from './drag-handle';
+import {DOCUMENT} from '@angular/platform-browser';
+import {CdkDropContainer, CDK_DROP_CONTAINER} from './drop-container';
+import {supportsPassiveEventListeners} from '@angular/cdk/platform';
+import {CdkDragStart, CdkDragEnd, CdkDragExit, CdkDragEnter, CdkDragDrop} from './drag-events';
+import {CdkDragPreview} from './drag-preview';
+import {CdkDragPlaceholder} from './drag-placeholder';
+
+/** Event options that can be used to bind an active event. */
+const activeEventOptions = supportsPassiveEventListeners() ? {passive: false} : false;
+
+// TODO: add an API for moving a draggable up/down the
+// list programmatically. Useful for keyboard controls.
+
+/** Element that can be moved inside a CdkDrop container. */
+@Directive({
+  selector: '[cdkDrag]',
+  exportAs: 'cdkDrag',
+  host: {
+    'class': 'cdk-drag',
+  }
+})
+export class CdkDrag implements AfterContentInit, OnDestroy {
+  private _document: Document;
+
+  /** Element that is displayed next to the user's pointer while the element is being dragged. */
+  private _preview: HTMLElement;
+
+  /** Reference to the view of the preview element. */
+  private _previewRef: EmbeddedViewRef<any> | null;
+
+  /** Reference to the view of the placeholder element. */
+  private _placeholderRef: EmbeddedViewRef<any> | null;
+
+  /** Element that is rendered instead of the draggable item while it is being sorted. */
+  private _placeholder: HTMLElement;
+
+  /** Coordinates within the element at which the user picked up the element. */
+  private _pickupPositionInElement: Point;
+
+  /** Coordinates on the page at which the user picked up the element. */
+  private _pickupPositionOnPage: Point;
+
+  /** CSS `transform` applied to the element when it isn't being dragged.  */
+  private _passiveTransform: Point = {x: 0, y: 0};
+
+  /** CSS `transform` that is applied to the element while it's being dragged. */
+  private _activeTransform: Point = {x: 0, y: 0};
+
+  /** Whether the element is being dragged. */
+  private _isDragging = false;
+
+  /** Whether the element has been moved since the user started dragging it. */
+  private _hasMoved = false;
+
+  /** Drop container that the element was in at the time it was picked up. */
+  private _initialContainer: CdkDropContainer;
+
+  /** Element that can be used to drag the draggable item. */
+  @ContentChild(CdkDragHandle) _handle: CdkDragHandle;
+
+  /** Element that will be used as a template to create the draggable item's preview. */
+  @ContentChild(CdkDragPreview) _previewTemplate: CdkDragPreview;
+
+  /**
+   * Element that will be used as a template to create a placeholder
+   * for the draggable item while it's being dragged.
+   */
+  @ContentChild(CdkDragPlaceholder) _placeholderTemplate: CdkDragPlaceholder;
+
+  /** Emits when the user starts dragging the item. */
+  @Output() started = new EventEmitter<CdkDragStart>();
```

Comment:
`@Output('cdkDragStarted') started` and similar for other outputs since this is an attribute selector directive